subprojects/language-java/src/integTest/groovy/org/gradle/javadoc/JavadocIntegrationTest.groovy

```diff
@@ -61,6 +62,30 @@ class JavadocIntegrationTest extends AbstractIntegrationSpec {
         file("build/docs/javadoc/Foo.html").text.contains("""Hey Joe!""")
     }
 
+    @Issue("gradle/gradle#1090")
+    @Unroll
+    def "allow single quote characters in options when string delimited with #delimiter"() {
+        given:
+        buildFile << """
+            apply plugin: "java"
+            javadoc.options.header = ${delimiter}${header}${delimiter}
+        """
+
+        file("src/main/java/Foo.java") << "public class Foo {}"
+
+        when:
+        succeeds 'javadoc'
+
+        then:
+        file("build/docs/javadoc/Foo.html").text.contains('"\'header\'"')
+
+        where:
+        delimiter | header
```

Comment:
Sorry, I still don't fully understand the test case we are running though here. Can you please explain? Maybe should just have the simplest test case that was actually reported: the use of single quote characters. The other two test case I do not get. For example when would you want to use "/"? The other test case is just verifying that Groovy can use single or double quotes to define a String.